### PR TITLE
Search improvements

### PR DESCRIPTION
**Pull request by MichielDeMey**
Improved search algorithm
Sorting alphabetically (secondary sort)
Throwing an error when the search query is too small

### CODE REVIEW

File: db/stations.js [hunk 1]
```diff
@@ -50,8 +50,21 @@ exports.levenLookup = function(needle, limit, callback) {
...
-        results.push({ score: score, id: parseInt(id), station: station });
+
+      // manipulate the score for stations beginning with the specified needle
+      // e.g gent -> gent-sint-pieters, gent-dampoort, ...
+      if (station.indexOf(needle) === 0) {
+        score = 0;
+      }
+
+      // if the score is larger than 5, it's probably nonsense
+      if (score < 5 ) {
```
Comment (gillesdemey) [inline db/stations.js]: It probably makes more sense to return from the iteration when the score is larger than 5.

```javascript
var score = leven(needle, station);
if (score > 5 ) // if the score is larger than 5, it's probably nonsense
  return;
```

Comment (MichielDeMey) [inline db/stations.js]: Makes sense, I'll change it.

Comment (gillesdemey) [inline db/stations.js]: On second thought, it's fine like this.

**Comment by gillesdemey** [comment]
:+1:

**Pull request merged**